core/src/test/java/org/ehcache/EhcacheBasicPutAllEventsTest.java

```diff
@@ -0,0 +1,299 @@
+
+/**
+ * @author
+ */
+@SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+public class EhcacheBasicPutAllEventsTest extends EhcacheBasicCrudBase{
+
+    @Mock
+    protected CacheLoaderWriter<String, String> cacheLoaderWriter;
+
+
+    @Mock
+    protected CacheEventListener testCacheEventListener;
+
+    protected CacheEventNotificationService cacheEventNotificationService;
+
+    protected IsCreatedOrUpdated isCreatedOrUpdated = new IsCreatedOrUpdated();
+
+    /**
+     * A Mockito {@code ArgumentCaptor} for the {@code Set} argument to the
+     * {@link org.ehcache.spi.cache.Store#bulkCompute(java.util.Set, org.ehcache.function.Function, org.ehcache.function.NullaryFunction)
+     *    Store.bulkCompute(Set, Function, NullaryFunction} method.
+     */
+    @Captor
+    private ArgumentCaptor<Set<String>> bulkComputeSetCaptor;
+
+    /**
+     * A Mockito {@code ArgumentCaptor} for the
+     * {@link org.ehcache.exceptions.BulkCacheWritingException BulkCacheWritingException}
+     * provided to the
+     * {@link org.ehcache.resilience.ResilienceStrategy#putAllFailure(java.util.Map, org.ehcache.exceptions.CacheAccessException, org.ehcache.exceptions.BulkCacheWritingException)}
+     *    ResilienceStrategy.putAllFailure(Iterable, CacheAccessException, BulkCacheWritingException)} method.
+     */
+    @Captor
+    private ArgumentCaptor<BulkCacheWritingException> bulkExceptionCaptor;
+
+
+    @Test
+    public void testPutAllNull() throws Exception {
+        final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+        final FakeStore fakeStore = new FakeStore(originalStoreContent);
+        this.store = spy(fakeStore);
+
+        final Ehcache<String, String> ehcache = this.getEhcache(null);
+        try {
+            ehcache.putAll(null);
+            fail();
+        } catch (NullPointerException e) {
+            // Expected
+        }
+        verify(testCacheEventListener,never()).onEvent(any(CacheEvent.class));
+    }
+
+
+    /**
+     * Tests {@link Ehcache#putAll(Map)} for
+     * <ul>
+     *    <li>non-empty request map</li>
+     *    <li>populated {@code Store} - some keys overlap request</li>
+     *    <li>populated {@code CacheLoaderWriter} - some keys overlap</li>
+     *    <li>no {@link CacheLoaderWriter#writeAll(Iterable)} calls fail</li>
+     * </ul>
+     */
+    @Test
+    public void testPutAll() throws Exception {
+        final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+        final FakeStore fakeStore = new FakeStore(originalStoreContent);
+        this.store = spy(fakeStore);
+
+        final Map<String, String> originalWriterContent = getEntryMap(KEY_SET_A, KEY_SET_B, KEY_SET_C);
+        final FakeCacheLoaderWriter fakeLoaderWriter = new FakeCacheLoaderWriter(originalWriterContent);
+        this.cacheLoaderWriter = spy(fakeLoaderWriter);
+
+        final Ehcache<String, String> ehcache = this.getEhcache(this.cacheLoaderWriter);
```

Comment:
There doesn't seem to be a need to provide a cacheLoaderWriter for this test case